lib/parallel_tests/cucumber/scenarios.rb

```diff
@@ -17,13 +17,71 @@ def all(files, options={})
 
         private
 
+        # Class stolen from cucumber as it was private
+        class NormalisedEncodingFile
+          COMMENT_OR_EMPTY_LINE_PATTERN = /^\s*#|^\s*$/ #:nodoc:
+          ENCODING_PATTERN = /^\s*#\s*encoding\s*:\s*([^\s]+)/ #:nodoc:
+
+          def self.read(path)
+            new(path).read
+          end
+
+          def initialize(path)
+            begin
+              @file = File.new(path)
+              set_encoding
+            rescue Errno::EACCES => e
+              raise FileNotFoundException.new(e, File.expand_path(path))
+            rescue Errno::ENOENT => e
+              raise FeatureFolderNotFoundException.new(e, path)
+            end
+          end
+
+          def read
+            @file.read.encode("UTF-8")
+          end
+
+          private
+
+          def set_encoding
+            @file.each do |line|
+              if ENCODING_PATTERN =~ line
+                @file.set_encoding $1
+                break
+              end
+              break unless COMMENT_OR_EMPTY_LINE_PATTERN =~ line
+            end
+            @file.rewind
+          end
+        end
+
         def split_into_scenarios(files, tags=[])
-          tag_expression = ::Gherkin::TagExpression.new(tags)
+          tag_expression = ::Cucumber::Core::Gherkin::TagExpression.new(tags)
           scenario_line_logger = ParallelTests::Cucumber::Formatters::ScenarioLineLogger.new(tag_expression)
-          loader = ::Cucumber::Runtime::FeaturesLoader.new(files, [], tag_expression)
 
-          loader.features.each do |feature|
-            feature.accept(scenario_line_logger)
+          features ||= files.map do |path|
```

Comment:
this looks like copy-pasted code too ... any chance we can reuse/patch something instead ?
... if not document what is copy-pasted and what is different ...